code/go/0chain.net/smartcontract/storagesc/stakepool.go

```diff
@@ -564,11 +567,12 @@ func (sp *stakePool) slash(allocID, blobID string, until common.Timestamp,
 	// related stake holders, that can loose some tokens due to
 	// division error;
 
-	var ap = wp.allocPool(allocID, until)
+	var ap = wp.allocPool(alloc.ID, until)
```

Comment:
Could we check `alloc` before access `alloc.id` ?.  App will crash when it is nil. It might be an unpredictable crash on production.

Reply:
It is impossible for alloc to nil. I could put a `panic` in, but if alloc is `nil` there is an earlier coding error that `must` be fixed. It should `never` crash in production.